### PR TITLE
Fetching data from api

### PR DESCRIPTION
**Pull request by esraaAbd-Elfatah**
Fetching Home DataFrom Api
Adding Class ApiServices 
Adding Home ViewModel
Adding Response Model

### CODE REVIEW

File: iOS_Project/Module/Home/ViewModels/HomeViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    var fetchCategoriesDataToHomeViewController : (()->())={}
+    
+    var fetchHomeData:Categories!{
+        didSet{
+            fetchCategoriesDataToHomeViewController()
+        }
+    }
+    
+    func fetchHomeData(tag :String){
+        NetworkServices.fetchHomeCategoriesData(tag:tag) { res in
```
Comment (MinaKamal99) [inline iOS_Project/Module/Home/ViewModels/HomeViewModel.swift]: adding this as a dependency injection

File: iOS_Project/Module/Home/ViewModels/HomeViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
+//
+//  HomeViewModel.swift
+//  iOS_Project
+//
+//  Created by Esraa AbdElfatah on 27/05/2023.
+//
+
+import Foundation
+class HomeViewModel{
```
Comment (MinaKamal99) [inline iOS_Project/Module/Home/ViewModels/HomeViewModel.swift]: please add a protocol to be confirmed

File: iOS_Project/Module/Home/View/HomeViewController.swift [hunk 3]
```diff
@@ -15,7 +15,9 @@ class HomeViewController: UIViewController , UITableViewDelegate,UITableViewData
     override func viewDidLoad() {
         super.viewDidLoad()
         self.title = "Food Recipies"
-        
+//        NetworkServices.fetchHomeCategoriesData(tag: "breakfast"){ (result) in
```
Comment (MinaKamal99) [inline iOS_Project/Module/Home/View/HomeViewController.swift:18]: please delete this if we no need it any more

**Comment by esraaAbd-Elfatah** [comment]
Resolved

File: iOS_Project/Module/Home/View/HomeViewController.swift (hunk 3 above)
Comment (esraaAbd-Elfatah) [reply iOS_Project/Module/Home/View/HomeViewController.swift:18]: Resolved

File: iOS_Project/Module/Home/ViewModels/HomeViewModel.swift (hunk 1 above)
Comment (esraaAbd-Elfatah) [reply iOS_Project/Module/Home/ViewModels/HomeViewModel.swift]: Resolved

File: iOS_Project/Module/Home/ViewModels/HomeViewModel.swift (hunk 2 above)
Comment (esraaAbd-Elfatah) [reply iOS_Project/Module/Home/ViewModels/HomeViewModel.swift]: Resolved

**Review by MinaKamal99 — APPROVED** [review]
Thanks 🙏

**Pull request merged**